Source/SelfService/Web/microservice/microserviceView/configuration/setupSection.tsx

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { useSnackbar } from 'notistack';
+
+import { Accordion, AccordionDetails, AccordionSummary, Box, Typography } from '@mui/material';
+import { DeleteRounded, EditRounded, ExpandCircleDownRounded, SaveRounded, RestartAltRounded } from '@mui/icons-material';
+
+import { Button } from '@dolittle/design-system/atoms/Button';
+import { Form, Input, Select, SwitchToggle } from '@dolittle/design-system/atoms/Forms';
+
+import { canDeleteMicroservice, deleteMicroservice, MicroserviceStore } from '../../../stores/microservice';
+
+import { HttpResponseApplication } from '../../../api/application';
+import { getRuntimes } from '../../../api/api';
+
+import { microserviceRestart } from '../helpers';
+import { AlertDialog } from './AlertDialog';
+import { HeadArguments } from '../../components/headArguments';
+
+const styles = {
+    form: {
+        'mt': 3,
+        '& .MuiFormControl-root': {
+            'my': 1,
+            '& fieldset': {
+                borderStyle: 'dashed'
+            }
+        },
+        '.MuiFormControlLabel-root': {
+            ml: 0
+        }
+    },
+    formSections: {
+        mb: 4,
+        display: 'flex',
+        flexDirection: 'column'
+    }
+};
+
+const runtimeVersions: { value: string }[] = [];
+
+type SetupSectionProps = {
+    application: HttpResponseApplication;
+    applicationId: string;
+    environment: string;
+    microserviceId: string;
+    currentMicroservice: MicroserviceStore;
+};
+
+export type SetupSectionParameters = {
+    microserviceName: string;
+    developmentEnvironment: string;
+    imageName: string;
+    port: number;
+    entrypoint: string;
+    ingressPath: string;
+};
+
+export const SetupSection = ({ application, applicationId, environment, microserviceId, currentMicroservice }: SetupSectionProps) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const history = useHistory();
+
+    const environmentInfo = application.environments.find(_environment => _environment.name === environment)!;
+    const canDelete = canDeleteMicroservice(application.environments, environment, microserviceId);
+
+    const { edit: { extra: {
+        headCommand,
+        runtimeImage,
+        headImage,
+        ingress
+    } } } = currentMicroservice;
+
+    const [setupPanelExpanded, setSetupPanelExpanded] = useState(true);
+    const [deleteDialogIsOpen, setDeleteDialogIsOpen] = useState(false);
+    const [formIsNotEditable, setFormIsNotEditable] = useState(true);
+    const [hasPublicURL, setHasPublicURL] = useState(false);
+    const [selectedRuntimeImage, setSelectedRuntimeImage] = useState('');
+    const [hasM3Connector, setHasM3Connetcor] = useState(environmentInfo.connections.m3Connector || false);
```

Comment:
`environmentInfo.connections.m3Connector` - this value tells if an m3 connector is available to be connected to this microservice, and can be mapped to a derived variable called `canUseM3Connector`.

the state variable `hasM3Connector` can be renamed to `useM3Connector`.